### PR TITLE
Add lab a4

### PR DESCRIPTION
**Pull request by ethanhs**
Found another missing word. Proofreading works!

### CODE REVIEW

**Comment by ethanhs** [comment]
Note to self: use https://docs.google.com/forms/d/e/1FAIpQLSf5NES0-oVSGjASP1LlKvilJ4Zq0Juund-5FjEUKGU3l4_41A/viewform

File: labs/a4.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    specifically what happened, do `cat ~/.ssh/id_rsa.pub`) on `tsunami`
+    and then compare that to `cat ~/.ssh/authorized_keys` on your student VM.
+
+At the end `ssh-copy-id` will prompt you to attempt to log into the server 
+using your key. This should succeed. 
+
+_Checkoff: paste the contents of `~/.ssh/authorized_keys` from your 
+student VM._
+
+Checkoff bonus: configure `sshd` to only allow key-based login.
```
Comment (cg505) [inline labs/a4.md]: what is a "checkoff bonus"?

File: labs/a4.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+1. All incoming connections are blocked by default.
+2. Allow incoming connections to your VM over port 22 (SSH), otherwise you'll be
+   locked out of your VM (as will we).
+3. Allow incoming connections for the ports necessary for the operation of the
+   software you'll be installing later on in this lab, e.g. port 80 and/or 443
+   for a web server.
+4. Bonus points: Allow the above, but only for IPs originating from UC Berkeley's
+   subnet. UC Berkeley has 3 primary `/16`s.
+5. Double bonus points: Configure `fail2ban` to block IP addresses that are trying
```
Comment (cg505) [inline labs/a4.md]: same q as before... do we have actual bonus points? (no)

File: labs/a4.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+Instead of uploading your files to Google Drive or Dropbox to be shared, you can
+host on your own VM. A good option for this is [fluffy](https://github.com/chriskuehl/fluffy)
+(created by an OCF alum!), which also doubles as a pastebin for quickly sharing code snippets.
+For more options, look at the the [“simple click/drag-n-drop upload” section of
+[awesome-selfhosted](https://github.com/Kickball/awesome-selfhosted#single-clickdrag-n-drop-upload).
+
+### Password Manager
+
+[BitWarden](https://blog.bitwarden.com/host-your-own-open-source-password-manager-ace147649936) is
```
Comment (cg505) [inline labs/a4.md]: noice. perhaps add a short note that you should _not_ store your password on your student VM, as you will lose access at the end of the semester, and decal staff have access (although not to the password directly)

File: labs/a4.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+a popular self-hosted password manager. You can use it to host your passwords if you
+don’t want to rely on a proprietary third-party program (like LastPass). Alternatives include
+[pass](https://passwordstore.org) which is a user-friendly wrapper around `gpg`. 
+
+# Submission
+Respond tot he checkoff questions on gradescope!
```
Comment (cg505) [inline labs/a4.md]:
```suggestion
Respond to the checkoff questions on gradescope!
```

What are these questions going to be? If they are substantial, please put them in the lab as well.

File: labs/a4.md (hunk 1 above)
Comment (ethanhs) [reply labs/a4.md]: I should reword the "checkoff" wording, thanks.

File: labs/a4.md (hunk 4 above)
Comment (cg505) [reply labs/a4.md]: note
>What are these questions going to be? If they are substantial, please put them in the lab as well.

Comment (ethanhs) [reply labs/a4.md]: Ah, yeah I'll go through them and figure that out.

**Pull request merged**